lib/smack/SmackInstGenerator.cpp

```diff
@@ -324,12 +326,29 @@ void SmackInstGenerator::visitAllocaInst(llvm::AllocaInst& ai) {
 
 void SmackInstGenerator::visitLoadInst(llvm::LoadInst& li) {
   processInstruction(li);
-  const Expr* rhs = rep.mem(li.getPointerOperand());
 
-  if (rep.isFloat(&li))
-    rhs = Expr::fn("$si2fp", rhs);
+  if (rep.tryBitVector()) { 
+    if (rep.tryDSA()) {
```

Comment:
No need for such complicated branching, there are only two cases. You can simplify:
````
if (!rep.tryBitVector() || rep.tryDSA() && rep.isFieldDisjoint(...)) {
  emit(Stmt::assign(...));
} else {
  emit(rep.loadAsBytes(...));
}